_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,293 @@
+
+For using GoogleTest in your code, you need to follow the following steps in general.
+
+
+
+### 3.1. Adding the required header files
+
+The first step would be to add the required header files in your program. For Googletest, you would need to add the following line in your code.
+
+```cpp
+#include "gtest/gtest.h"
+```
+
+
+
+### 3.2. Create your tests
+
+The next step would be to to define your test. Googletest uses the following convention in naming various tests.
+
+```cpp
+TEST(TestSuiteName, TestName)
+{
+    // Test logic and assertions
+}
+```
+
+The different parts in the above cell have the following meaning
+
+- `TEST`: This is the macro provided by the Google Test framework to define a test case.
+- `TestSuiteName`: It is the name of the test suite or test fixture. It groups related tests together. It should be a valid C++ identifier.
+- `TestName`: It is the name of the individual test case within the test suite. It should also be a valid C++ identifier.
+- `Test logic and assertions`: This is the body of the test case where you write the actual test code, including any necessary assertions to verify the expected behaviour. The section `Test Assertions` describes more about various assertions available in GoogleTest.
+
+
+
+### 3.3. Initialise GoogleTest framework in your main
+
+For this sub-section, we are assuming that you are writing your own main function (GoogleTest also allows you to run your code without your own main function by linking to a main function provided in `lgest_main`. Please see the next section for more details). You would need to include the following lines in your main function as shown below.
+
+```cpp
+int main(int argc, char **argv)
+{
+    // Initialise GoogleTest Framework
+    testing::InitGoogleTest(&argc, argv);
+    
+    // Instruction to Run the test cases.
+    return RUN_ALL_TESTS();
+}
+```
+
+The `RUN_ALL_TESTS()` is typically called from `main` and when it is invoked, it scans the program for all the test cases and  test methods defined using the `test` macro.   It then executes each test case and captures the results of each individual test method within the test case. After running all the tests, it provides a summary of the test results, including the number of tests run, passed, and failed.
+
+
+
+### 3.4. Compile and run your program
+
+In order to compile your program and link with the required libraries, you can use the following instruction as a template. Please remember to modify the paths according to your system and location of the files.
+
+```bash
+# For compiling and linking, please use the instruction below
+$ g++ your_code.cpp -I path_to_your_gtest.h -L path_to_your_lgtest.a -lgtest -lpthread -o your_executable_name
+ 
+# Run your code  
+$ ./your_executable_name
+```
+
+If you do not want to write your main function, you can link against the one provided in the `libgtest_main.a` library by using the following instruction.
```

Comment:
As mentioned above, we need some context about where this library is coming from.

Reply:
A brief describe in cf42314